tests/sklearn/test_sklearn_autolog.py

```diff
@@ -1106,3 +1107,326 @@ def fit(self, X, y):
+    X, y = get_iris()
+    X_eval = X[:-1, :]
+    y_eval = y[:-1]
+
+    with mlflow.start_run() as run:
+        model.fit(X, y)
+        eval_run_id, eval_metrics, eval_artifacts = mlflow.sklearn.log_eval_metrics(
+            model=model, X=X_eval, y_true=y_eval, prefix="eval_"
+        )
+
+    # Check that the existing active run was used.
+    run_id = run.info.run_id
+    assert eval_run_id == run_id
+
+    # Check that the logged metrics/artifacts for the run are exactly those returned by the call
+    _, metrics, _, artifacts = get_run_data(run_id)
+    assert sorted(artifacts) == sorted(eval_artifacts)
+    assert metrics == eval_metrics
+
+    # Check the actual metrics and artifacts
+    y_pred = model.predict(X_eval)
+
+    # SVC does not support probability predictions so the corresponding metrics (log_loss, auc)
+    # are missing.
+    expected_metrics = {
+        "eval_score": model.score(X_eval, y_eval),
+        "eval_accuracy_score": sklearn.metrics.accuracy_score(y_eval, y_pred),
+        "eval_precision_score": sklearn.metrics.precision_score(y_eval, y_pred, average="weighted"),
+        "eval_recall_score": sklearn.metrics.recall_score(y_eval, y_pred, average="weighted"),
+        "eval_f1_score": sklearn.metrics.f1_score(y_eval, y_pred, average="weighted"),
+    }
+    assert eval_metrics == expected_metrics
+
+    client = mlflow.tracking.MlflowClient()
+    artifacts = [x.path for x in client.list_artifacts(run_id)]
+    plot_names = ["{}.png".format("eval_confusion_matrix")] if _is_plotting_supported() else []
+    assert plot_names == eval_artifacts
+
+
+def test_log_eval_metrics_with_new_run():
+    # disable autologging so that we can check for the sole existence of eval-time metrics
+    mlflow.sklearn.autolog(disable=True)
+
+    # use simple `LinearRegression`, which only implements `fit`.
+    model = sklearn.linear_model.LinearRegression()
+    X, y_true = get_iris()
+    X_eval = X[:-1, :]
+    y_eval = y_true[:-1]
+    model = fit_model(model, X, y_true, "fit")
+
+    run_id, eval_metrics, eval_artifacts = mlflow.sklearn.log_eval_metrics(
+        model=model, X=X_eval, y_true=y_eval, prefix="eval_"
+    )
+    # Check the contents for the metrics and artifacts
+    y_pred = model.predict(X_eval)
+    assert eval_metrics == {
+        "eval_score": model.score(X_eval, y_eval),
+        "eval_mse": sklearn.metrics.mean_squared_error(y_eval, y_pred),
+        "eval_rmse": np.sqrt(sklearn.metrics.mean_squared_error(y_eval, y_pred)),
+        "eval_mae": sklearn.metrics.mean_absolute_error(y_eval, y_pred),
+        "eval_r2_score": sklearn.metrics.r2_score(y_eval, y_pred),
+    }
+    assert len(eval_artifacts) == 0
+
+    # Check the the logged metrics/artifacts are the same as the returned ones.
+    _, metrics, _, artifacts = get_run_data(run_id)
+    assert eval_metrics == metrics
+    assert eval_artifacts == artifacts
+
+
+def test_log_eval_metrics_throws_with_invalid_args():
```

Comment:
I can think of the following cases:
- A non-sklearn model.
- A model_uri that fails to resolve to a valid model.

Did you have other cases in mind?

Reply:
Are there sklearn models without `score()`? If not, I think testing a non-sklearn model should be sufficient.

Reply:
The sklearn documentation suggests that all Estimators have a `score` although I found this to be wrong (e.g., some clusterers inherit from BaseEstimator but do not support neither `score` nor `predict`). However, we don't care about the score method since we check for it before recording any metrics. I added a test to exercise this behavior. Instead, we care for `predict` so I added this to the list of requirements for `model`. Let me know if this works.